### PR TITLE
Script docs

### PR DESCRIPTION
**Pull request by romseygeek**
This commit adds some more details and examples to the docs for the `score`, `weight`, `similarity` and `sort` painless contexts.

Relates to #34829

### CODE REVIEW

**Comment by romseygeek** [comment]
I haven't added examples for `weight` or `similarity` because they don't really fit with the painless example dataset, and the docs for `scripted_similarity` cover that pretty well.

**Review by nik9000 — APPROVED** [review]

File: docs/painless/painless-contexts/painless-score-context.asciidoc [hunk 1]
```diff
@@ -24,4 +26,32 @@ score to documents returned from a query.
...
+*Example*
+
+To run this example, first follow the steps in
+<<painless-context-examples, context examples>>.
+
+The following query will find all unsold seats, with lower 'row' values
+scored higher.
+
+[source,js]
+--------------------------------------------------
```
Comment (nik9000) [inline docs/painless/painless-contexts/painless-score-context.asciidoc]: It looks like this one is failing when we run the docs tests. I expect the index is missing.

**Review by nik9000 — CHANGES_REQUESTED** [review]

File: docs/painless/painless-contexts/painless-score-context.asciidoc (hunk 1 above)
Comment (romseygeek) [reply docs/painless/painless-contexts/painless-score-context.asciidoc]: This should be fixed now

**Comment by romseygeek** [comment]
Test failure is an unrelated CCR one - given that the docs build has passed, I think this is good to go?

**Comment by jdconrad** [comment]
@romseygeek Asked @debadair to take a look as well from the doc side of things.  Thanks for doing this!

**Review by debadair — APPROVED** [review]
Just a few editorial nits.

**Review by jdconrad — APPROVED** [review]
LGTM!  Thanks for doing this!  @nik9000 Are you happy with the changes now?

**Review by nik9000 — APPROVED** [review]
LGTM

**Pull request merged**